firmware/shared/controls/DI_driverTorqueRequest.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <cmath>
+#include <iostream>
+
+namespace ctrl {  
+
+// Templating means this function will work on any types that support addition.
+template <typename T, typename B>

```

Comment:
We expect READY_TO_DRIVE and DI_Error to be booleans, so they should not be templated. The accelpedal pos should be templated as you have it.